netlogo-gui/src/main/app/InterfaceTab.scala

```diff
@@ -78,14 +80,16 @@ class InterfaceTab(workspace: GUIWorkspace,
     if(iP.isFocusable && splitPane.getDividerLocation >= maxDividerLocation) {
       iP.requestFocus
     }
-    else {
-      lastFocusedComponent.requestFocus
+    else if(InterfaceTab.interfacePanelHadFocus){
+      iP.requestFocus
+    } else {
+      commandCenter.requestFocus()
     }
   }
 
   final def handle(e: Events.SwitchedTabsEvent) {
     if(e.newTab != this) {
-        lastFocusedComponent = if(iP.hasFocus) iP else commandCenter
```

Comment:
I really prefer this `lastFocusedComponent` approach to the `InterfaceTab.interfacePanelHadFocus` approach. If you added a method to `InterfacePanel` to make `_hasFocus` accessible and used that here instead of `hasFocus`, could you make this approach work? If that won't work, could you have `InterfaceTab` add a `FocusListener` to each of `CommandCenter` and `InterfacePanel` and have the listener(s) change an instance variable in `InterfaceTab`?

Basically, we want to try every possible alternative approach before adding a public static variable like `InterfaceTab.interfacePanelHadFocus`. Public static variables make it really hard to debug code and really hard to change code. If neither of the above approaches work, please let me know and we'll see if we can come up with another alternative.